src/coreComponents/finiteVolume/ProjectionEDFMHelper.hpp

```diff
@@ -0,0 +1,94 @@
+#ifndef __PROJECTIONEDFMHELPER_H_
+#define __PROJECTIONEDFMHELPER_H_
+
+#include "finiteVolume/FluxApproximationBase.hpp"  // FIXME: include appropriate explicitly
+#include <vector>                                  // provides std::vector
+
+namespace geosx {
+
+class ProjectionEDFMHelper {
+ public:
+  ProjectionEDFMHelper( MeshLevel const & mesh,
+                        GeometricObjectManager const * geometricObjManager,
+                        std::string const & coeffName,
+                        CellElementStencilTPFA & stencil );
+
+  // add Fracture-matrix connections to the cell stencil
+  void addNonNeighboringConnections(EmbeddedSurfaceSubRegion const & fractureSubRegion);
```

Comment:
I wonder if instead of creating a `ProjectionEDFMHelper` each time, it would be better to make it inherit from `Group` and register it as a `subGroup` of the `TwoPointFluxApproximation`. We could then also add a method to update the stencils whenever the aperture changes by grabbing it from the `TwoPointFluxApproximation`. 
To do so you would have to change the constructor though and populate the object in a separate method because I think that when the `TwoPointFluxApproximation` is generated a lot of the info needed is not available yet. 

@rrsettgast, @francoishamon, @klevzoff, what do you think?

Reply:
Yes I agree with @cusiniM. If these helpers can be used in multiple places in the code and can be called every time the aperture changes, then it is worth trying to make the object persistent by registering a `ProjectionEDFMHelper` on the `TwoPointFluxApproximation` object. 
- Then, once all the views that you need in the helpers have been created, you would populate the registered object and copy all the views into the data members of `ProjectionEDFMHelper`. Then you would grab the `ProjectionEDFMHelper` object, and just call `addNonNeighboringConnections` and or the additional function that @cusiniM is mentioning.
- Another possibility is to remove all the data members from `ProjectionEDFMHelper` and just extend the list of input parameters in `addNonNeighboringConnections`. You would have to pass `mesh`, `geometricObjManager`, and `stencil` in addition to `fractureSubRegion` in the call to `addNonNeighboringConnections`. With this option you would not have to worry about populating the `ProjectionEDFMHelper` object.

If the object `ProjectionEDFMHelper` is not registered on the `TwoPointFluxApproximation` but still needs to be used in multiple places, there is always the option to make the functions in the class `ProjectionEDFMHelper` `static` and pass them all the parameters that they need everytime they are called. 

That being said, I am not sure about the best option here. If the `mesh`, `geometricObjManager`, and `stencil` objects are still available when/where you want to update the stencils after the aperture change, maybe there is no need to store them in the class?

Reply:
I wouldn't go as far as making `ProjectionEDFMHelper` inherit from `Group` because it doesn't actually allocate and permanently store any data, just grabs some views to use. On the other hand, to grab those views it does a few key lookups, so we do want to cache those lookups within the object (with the caveat that views may get invalidated if the underlying arrays are resized - make sure this is not the case here).

I think the way @ishovkun has gone is to create a subobject of `FluxApproximation` (managed through a smart pointer, so it can be created optionally) but not register it as a subgroup. Seems like a reasonable compromise?

Reply:
The initialization of ProjectionEDFMHelper is moved to TwoPointFluxApproximation::initializePreSubgroups (775d1d0).

@klevzoff, @francoishamon, @CusiniM Please let me know what y'all think.